### PR TITLE
Fix the username in the bot's response when the username is not set

### PR DESCRIPTION
**Pull request by alexkart**
When a user doesn't have username the bot's response looks like this:
```
Слишком много запросов, следующий запрос можно будет сделать через 19 минут.
@ получает бан на 1 час.
```

It will display the user's display name in this case.

### CODE REVIEW

File: app/bot/bot.go [hunk 1]
```diff
@@ -252,11 +252,18 @@ func EscapeMarkDownV1Text(text string) string {
...
 }
 
-// DisplayName returns user's display name or username or id
-func DisplayName(msg Message) string {
-	displayUsername := msg.From.DisplayName
-	if displayUsername == "" {
-		displayUsername = msg.From.Username
+// DisplayName returns user's display name or username or id, if usernameFirst is true it will try to get username first
+// instead of display name
+func DisplayName(msg Message, usernameFirst bool) string {
```
Comment (umputun) [inline app/bot/bot.go:257]: this boolean parameter is very confusing. Reading the consumer side with true/false requires IDE help to figure out what this unexpected boolean even means. Probably making it an enum (iota) type with descriptive values will help. Or even better, making two functions with distinct names would be a better approach.

**Review by umputun — COMMENTED** [review]
In addition to the comment above, the change is missing any tests. It would be helpful to explain why this is even a problem. The issue may be that multiple users with empty names are being considered as the same one, but this is not entirely clear.

**Comment by alexkart** [comment]
> The issue may be that multiple users with empty names are being considered as the same one, but this is not entirely clear.

I'm not sure I understand what you mean by this. The only issue here is that the bot's response looks like this `@ получает бан на 1 час.` instead of this `John Doe получает бан на 1 час.` when user doesn't set their username which is optional. Or did you mean something else?

**Comment by umputun** [comment]
Well, if this is the only display issue for users banned properly, I suggest minimizing the change and making it as local as possible in that context. Currently, it seems to affect too much for such a minor issue. If it is hard to localize, I would say - leave it as is, as it is not a big problem anyway and I prefer to have an empty name reported rather than something unexpected broken.

**Comment by alexkart** [comment]
> rather than something unexpected broken

Do you mean if there are some weird characters in the user's display name? But users' display names are already used in bot's responses like this:
```
this is spam! go to ban, "Bogdan Grebnev" (id:1951519279)
```
So, should we leave it as is or make a change but as local as possible, as you suggested?

**Comment by umputun** [comment]
Currently, you had to introduce a special mode for the display name, changing what you pass it and so on. I have not checked what else it can affect, but it feels like too much for what it is. If you can localize it and add a test, we can proceed. But if this is too much work, we can live without it. It's up to you.

cc @paskal

**Comment by alexkart** [comment]
> If you can localize it and add a test, we can proceed.

OK, I'll try.

> Well, if this is the only display issue for users banned properly
Actually, I'm not sure about it, how an actual ban works, I haven't looked at this functionality yet. Does it also ban users by username or by ID or something else? As I said, I've just tried to fix the display issue here, maybe the actual ban functionality should also be checked to see if it works correctly for users without usernames.

**Comment by alexkart** [comment]
I've localized the changes (created a separate function) and added tests. This PR is ready for review.

**Review by paskal — APPROVED** [review]

**Comment by paskal** [comment]
It looks like a pre-requisite for correctly displaying whatever bot cannot ban. It breaks nothing and improves something.

**Pull request merged**